contracts/ERC2470.sol

```diff
@@ -0,0 +1,71 @@
+// SPDX-License-Identifier: CC0-1.0
+pragma solidity 0.7.3;
+
+
+/**
+ * @title ERC-2470: Singleton Factory
+ * @notice Exposes CREATE2 (EIP-1014) to deploy bytecode on deterministic addresses based on initialization code and salt.
+ * @author Ricardo Guilherme Schmidt <ricardo3@status.im>
+ * @author Richard Meissner <richard@gnosis.io>
+ */
+contract ERC2470 {
+
+    /**
+     * @notice Fails on calls with no `msg.data` but with `msg.value`
+     */
+    receive() external payable {
+        revert("ERC2470: bad call");
+    }
+
+    /**
+     * @notice Fallback forwards `msg.data` with salt `bytes32(0)` to `deploy(bytes,bytes32)`.
+     */
+    fallback() external payable {
+        address payable createdContract = deploy(msg.data, bytes32(0));
+        assembly {
```

Comment:
I would add a comment for this (e.g. "Return address of created contract even for fallback"), just be make people aware what is going on and why assembly is needed

Reply:
Ok I added a comment at @dev, should be enough, wdyt?